### ISSUE TITLE
Initialization population cannot survive

### ISSUE THREAD

**Issue by pancak3** [msu-coinlab/pymoo#137]
I am playing around with NSGA2 and NSGA3. 
I read the documentation which says use `sampling` as the parameter to initialize with a defined population. 

Based on my test, the initial population successfully goes into the evaluation as the first population. 
However, none of them goes to the second population.
This should not happen as they are almost the optimal chromosomes.

Am I doing correctly to initialize with parameter `sampling`?  
And why does none of them survive even in the second generation?

Thanks for your kindly help!

--------------

I read the paper and found the elitism of the initial population should actually not go into the next generation.
My previous understanding was wrong.

**Comment by julesy89** [comment]
I see you probably have resolved your problem. What was the issue?

**Comment by pancak3** [comment]
I tried to customize my initial population of NSGA2.
And I thought the good chromosome will go into the second population, like a normal GA.
However, they never go.

Before I close this issue, I thought I initialize incorrectly or your code works incorrectly.
But after I read the NSGA2 paper, I found my understanding was wrong.
The initial population would actually not go into the next generation.
So I closed this issue.

All in all, your implementation works well!

If I have to advise, I would say the `do` methods are everywhere and it is hard to trace the code without debugging.

If you don't mind, I would like to add type hints after I finish my work and create a pr.
